NOnion/Network/TorStream.fs

```diff
@@ -10,17 +10,115 @@ open NOnion
 open NOnion.Cells.Relay
 open NOnion.Utility
 
-
 type TorStream(circuit: TorCircuit) =
 
     let mutable streamState: StreamState = StreamState.Initialized
     let controlLock: SemaphoreLocker = SemaphoreLocker()
 
     let window: TorWindow = TorWindow Constants.DefaultStreamLevelWindowParams
 
+    let mutable currentBuffer: array<byte> =
+        Array.zeroCreate Constants.MaximumRelayPayloadLength
+
+    let mutable bufferOffset: int = 0
+    let mutable bufferLength: int = 0
+    let mutable isEOF: bool = false
+
     let incomingCells: BufferBlock<RelayData> = BufferBlock<RelayData>()
     let receiveLock: SemaphoreLocker = SemaphoreLocker()
 
+    let CurrentBufferHasRemainingBytes() =
+        bufferLength > bufferOffset
+
+    let CurrentBufferRemainingBytes() =
+        bufferLength - bufferOffset
```

Comment:
why don't you make this return an Option, so that you don't need the function CurrentBufferHasRemainingBytes?